Survey/Sources/Domain/UseCases/CachedStorageUseCase.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  CachedStorageUseCase.swift
+//  Survey
+//
+//  Created by Khanh on 07/12/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+protocol CachedStorageUseCaseProtocol {
+
+    func store(data: [Survey])
+    func load() -> [Survey]
+    func delete() -> Observable<Bool>
+}
+
+final class CachedStorageUseCase: CachedStorageUseCaseProtocol {
```

Comment:
Please take a look at this [document](https://developer.android.com/topic/architecture/data-layer/offline-first#modeling-data). The network vs cached should be managed on the data layer, ideally by the same entity.
Current approach will have the presentation layer manage the network vs cached logic which can create a lot of duplicate code (as seen in this PR when adding the new usecase in every ViewModel), please review this approach.

Reply:
@blyscuit I fixed it at https://github.com/nkhanh44/ic-survey-ios/pull/85/commits/df6c41dae40f77dbbdb5a171133da7f2a2438088 by creating a repository, which belongs to the data layer, to handle cached. Please help me to review it again. Thank you 🙇 .